Continue porting WebCore/platform types to the new IPC serialization format

gavin-apple opened the pull request:
#### 32f6c4d6e879210e00c6383497cdb9c43bb8f36b
<pre>
Continue porting WebCore/platform types to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=251386">https://bugs.webkit.org/show_bug.cgi?id=251386</a>
rdar://104833997

Reviewed by Alex Christensen.

This change includes porting for types:
    - WebCore::LinearTimingFunction
    - WebCore::CubicBezierTimingFunction::TimingFunctionPreset
    - WebCore::CubicBezierTimingFunction
    - WebCore::StepsTimingFunction
    - WebCore::TextCheckingResult
    - WebCore::AudioIOPosition
    - WebCore::VideoFrameRotation
    - WebCore::VideoFrameCV
    - WebCore::GradientColorStop
    - WebCore::GradientColorStops
    - WebCore::ImageBufferBackendParameters
    - WebCore::PreserveResolution
    - WebCore::PathElementType
    - WebCore::EncodedPathElementType
    - WebCore::ProcessIdentity
    - WebCore::ColorInterpolationColorSpace
    - WebCore::HueInterpolationMethod
    - WebCore::ColorInterpolationMethod::HSL
    - WebCore::ColorInterpolationMethod::HWB
    - WebCore::ColorInterpolationMethod::LCH
    - WebCore::ColorInterpolationMethod::Lab
    - WebCore::ColorInterpolationMethod::OKLCH
    - WebCore::ColorInterpolationMethod::OKLab
    - WebCore::ColorInterpolationMethod::SRGB
    - WebCore::ColorInterpolationMethod::SRGBLinear
    - WebCore::ColorInterpolationMethod::XYZD50
    - WebCore::ColorInterpolationMethod::XYZD65
    - WebCore::ColorInterpolationMethod
    - WebCore::RouteSharingPolicy
    - WebCore::AudioSessionCategory
    - WebCore::AudioSession::MayResume
    - WebCore::AudioSessionRoutingArbitrationError
    - WebCore::AudioSessionRoutingArbitrationClient::DefaultRouteChanged
    - WebCore::CDMKeyStatus
    - WebCore::CDMMessageType
    - WebCore::CDMRequirement
    - WebCore::MediaConstraintType
    - WebCore::RTCDataChannelState
    - WebCore::RTCPriorityType
    - WebCore::HTTPCookieAcceptPolicy
    - WebCore::NetworkLoadPriority
    - WebCore::PrivacyStance
    - WebCore::ThirdPartyCookieBlockingMode
    - WebCore::FirstPartyWebsiteDataRemovalMode
    - WebCore::FontRenderingMode
    - WebCore::DragApplicationFlags
    - WebCore::TextCheckingType

* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCUtils.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::markAndReplaceFor):
* Source/WebCore/platform/DragData.h:
* Source/WebCore/platform/ProcessIdentity.h:
(WebCore::ProcessIdentity::encode const): Deleted.
(WebCore::ProcessIdentity::decode): Deleted.
* Source/WebCore/platform/VideoFrame.h:
* Source/WebCore/platform/animation/TimingFunction.h:
* Source/WebCore/platform/audio/AudioIOCallback.h:
(WebCore::AudioIOPosition::encode const): Deleted.
(WebCore::AudioIOPosition::decode): Deleted.
* Source/WebCore/platform/audio/AudioSession.h:
* Source/WebCore/platform/encryptedmedia/CDMKeyStatus.h:
* Source/WebCore/platform/encryptedmedia/CDMMessageType.h:
* Source/WebCore/platform/encryptedmedia/CDMRequirement.h:
* Source/WebCore/platform/graphics/ColorInterpolationMethod.h:
(WebCore::ColorInterpolationMethod::encode const): Deleted.
(WebCore::ColorInterpolationMethod::decode): Deleted.
* Source/WebCore/platform/graphics/GradientColorStop.h:
(WebCore::GradientColorStop::encode const): Deleted.
(WebCore::GradientColorStop::decode): Deleted.
* Source/WebCore/platform/graphics/GradientColorStops.h:
(WebCore::GradientColorStops::encode const): Deleted.
(WebCore::GradientColorStops::decode): Deleted.
* Source/WebCore/platform/graphics/ImageBufferBackend.h:
(WebCore::ImageBufferBackend::Parameters::encode const): Deleted.
(WebCore::ImageBufferBackend::Parameters::decode): Deleted.
* Source/WebCore/platform/graphics/ImageBufferBackendParameters.h: Copied from Source/WebCore/platform/network/HTTPCookieAcceptPolicy.h.
* Source/WebCore/platform/graphics/Path.h:
(WebCore::EncodedPathElementType::EncodedPathElementType):
(WebCore::EncodedPathElementType::isEndOfPath const):
(WebCore::EncodedPathElementType::isValidPathElementType const):
(WebCore::EncodedPathElementType::isValid const):
(WebCore::EncodedPathElementType::type const):
(WebCore::Path::EncodedPathElementType::EncodedPathElementType): Deleted.
(WebCore::Path::EncodedPathElementType::isEndOfPath const): Deleted.
(WebCore::Path::EncodedPathElementType::isValidPathElementType const): Deleted.
(WebCore::Path::EncodedPathElementType::isValid const): Deleted.
(WebCore::Path::EncodedPathElementType::type const): Deleted.
(WebCore::Path::EncodedPathElementType::encode): Deleted.
(WebCore::Path::EncodedPathElementType::decode): Deleted.
* Source/WebCore/platform/graphics/cv/VideoFrameCV.h:
(WebCore::VideoFrameCV::encode const): Deleted.
(WebCore::VideoFrameCV::decode): Deleted.
* Source/WebCore/platform/mediastream/MediaStreamRequest.h:
* Source/WebCore/platform/mediastream/RTCDataChannelState.h:
* Source/WebCore/platform/mediastream/RTCPriorityType.h:
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSupportedConstraints.h:
* Source/WebCore/platform/network/HTTPCookieAcceptPolicy.h:
* Source/WebCore/platform/network/NetworkLoadMetrics.h:
* Source/WebCore/platform/network/NetworkStorageSession.h:
* Source/WebCore/platform/network/cf/CertificateInfo.h:
* Source/WebCore/platform/text/FontRenderingMode.h:
* Source/WebCore/platform/text/TextChecking.h:
* Source/WebCore/platform/text/TextFlags.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.messages.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteVideoFrameProxy.h:

Canonical link: <a href="https://commits.webkit.org/259829@main">https://commits.webkit.org/259829@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f300f80fae98de5a24465aeb7c18578c9bbf3eb6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106063 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15118 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38894 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115244 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/175337 "Build was cancelled. Recent messages:Pull request contains relevant changes; Deleted stale build files; Encountered some issues during cleanup") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109965 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16544 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6288 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98271 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/114964 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111816 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12593 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95570 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/40141 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94461 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27213 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8369 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28565 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8861 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5114 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14482 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48109 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6800 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10405 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/editing/Editor.cpp:
```diff
@@ -2894,7 +2894,9 @@ void Editor::markAndReplaceFor(const SpellCheckRequest& request, const Vector<Te
         auto spellingRangeEndOffset = paragraph.checkingEnd() + offsetDueToReplacement;
-        TextCheckingType resultType = results[i].type;
+        if (!results[i].type.hasExactlyOneBitSet())
+            continue;
```
Probably add an ASSERT_NOT_REACHED here.  If regular code hits this, we want an engineer to notice it with an assertion.

gavin-apple committed:
Continue porting WebCore/platform types to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=251386
rdar://104833997

Reviewed by Alex Christensen.

This change includes porting for types:
    - WebCore::LinearTimingFunction
    - WebCore::CubicBezierTimingFunction::TimingFunctionPreset
    - WebCore::CubicBezierTimingFunction
    - WebCore::StepsTimingFunction
    - WebCore::TextCheckingResult
    - WebCore::AudioIOPosition
    - WebCore::VideoFrameRotation
    - WebCore::VideoFrameCV
    - WebCore::GradientColorStop
    - WebCore::GradientColorStops
    - WebCore::ImageBufferBackendParameters
    - WebCore::PreserveResolution
    - WebCore::PathElementType
    - WebCore::EncodedPathElementType
    - WebCore::ProcessIdentity
    - WebCore::ColorInterpolationColorSpace
    - WebCore::HueInterpolationMethod
    - WebCore::ColorInterpolationMethod::HSL
    - WebCore::ColorInterpolationMethod::HWB
    - WebCore::ColorInterpolationMethod::LCH
    - WebCore::ColorInterpolationMethod::Lab
    - WebCore::ColorInterpolationMethod::OKLCH
    - WebCore::ColorInterpolationMethod::OKLab
    - WebCore::ColorInterpolationMethod::SRGB
    - WebCore::ColorInterpolationMethod::SRGBLinear
    - WebCore::ColorInterpolationMethod::XYZD50
    - WebCore::ColorInterpolationMethod::XYZD65
    - WebCore::ColorInterpolationMethod
    - WebCore::RouteSharingPolicy
    - WebCore::AudioSessionCategory
    - WebCore::AudioSession::MayResume
    - WebCore::AudioSessionRoutingArbitrationError
    - WebCore::AudioSessionRoutingArbitrationClient::DefaultRouteChanged
    - WebCore::CDMKeyStatus
    - WebCore::CDMMessageType
    - WebCore::CDMRequirement
    - WebCore::MediaConstraintType
    - WebCore::RTCDataChannelState
    - WebCore::RTCPriorityType
    - WebCore::HTTPCookieAcceptPolicy
    - WebCore::NetworkLoadPriority
    - WebCore::PrivacyStance
    - WebCore::ThirdPartyCookieBlockingMode
    - WebCore::FirstPartyWebsiteDataRemovalMode
    - WebCore::FontRenderingMode
    - WebCore::DragApplicationFlags
    - WebCore::TextCheckingType

* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCUtils.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::markAndReplaceFor):
* Source/WebCore/platform/DragData.h:
* Source/WebCore/platform/ProcessIdentity.h:
(WebCore::ProcessIdentity::encode const): Deleted.
(WebCore::ProcessIdentity::decode): Deleted.
* Source/WebCore/platform/VideoFrame.h:
* Source/WebCore/platform/animation/TimingFunction.h:
* Source/WebCore/platform/audio/AudioIOCallback.h:
(WebCore::AudioIOPosition::encode const): Deleted.
(WebCore::AudioIOPosition::decode): Deleted.
* Source/WebCore/platform/audio/AudioSession.h:
* Source/WebCore/platform/encryptedmedia/CDMKeyStatus.h:
* Source/WebCore/platform/encryptedmedia/CDMMessageType.h:
* Source/WebCore/platform/encryptedmedia/CDMRequirement.h:
* Source/WebCore/platform/graphics/ColorInterpolationMethod.h:
(WebCore::ColorInterpolationMethod::encode const): Deleted.
(WebCore::ColorInterpolationMethod::decode): Deleted.
* Source/WebCore/platform/graphics/GradientColorStop.h:
(WebCore::GradientColorStop::encode const): Deleted.
(WebCore::GradientColorStop::decode): Deleted.
* Source/WebCore/platform/graphics/GradientColorStops.h:
(WebCore::GradientColorStops::encode const): Deleted.
(WebCore::GradientColorStops::decode): Deleted.
* Source/WebCore/platform/graphics/ImageBufferBackend.h:
(WebCore::ImageBufferBackend::Parameters::encode const): Deleted.
(WebCore::ImageBufferBackend::Parameters::decode): Deleted.
* Source/WebCore/platform/graphics/ImageBufferBackendParameters.h: Copied from Source/WebCore/platform/network/HTTPCookieAcceptPolicy.h.
* Source/WebCore/platform/graphics/Path.h:
(WebCore::EncodedPathElementType::EncodedPathElementType):
(WebCore::EncodedPathElementType::isEndOfPath const):
(WebCore::EncodedPathElementType::isValidPathElementType const):
(WebCore::EncodedPathElementType::isValid const):
(WebCore::EncodedPathElementType::type const):
(WebCore::Path::EncodedPathElementType::EncodedPathElementType): Deleted.
(WebCore::Path::EncodedPathElementType::isEndOfPath const): Deleted.
(WebCore::Path::EncodedPathElementType::isValidPathElementType const): Deleted.
(WebCore::Path::EncodedPathElementType::isValid const): Deleted.
(WebCore::Path::EncodedPathElementType::type const): Deleted.
(WebCore::Path::EncodedPathElementType::encode): Deleted.
(WebCore::Path::EncodedPathElementType::decode): Deleted.
* Source/WebCore/platform/graphics/cv/VideoFrameCV.h:
(WebCore::VideoFrameCV::encode const): Deleted.
(WebCore::VideoFrameCV::decode): Deleted.
* Source/WebCore/platform/mediastream/MediaStreamRequest.h:
* Source/WebCore/platform/mediastream/RTCDataChannelState.h:
* Source/WebCore/platform/mediastream/RTCPriorityType.h:
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSupportedConstraints.h:
* Source/WebCore/platform/network/HTTPCookieAcceptPolicy.h:
* Source/WebCore/platform/network/NetworkLoadMetrics.h:
* Source/WebCore/platform/network/NetworkStorageSession.h:
* Source/WebCore/platform/network/cf/CertificateInfo.h:
* Source/WebCore/platform/text/FontRenderingMode.h:
* Source/WebCore/platform/text/TextChecking.h:
* Source/WebCore/platform/text/TextFlags.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.messages.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteVideoFrameProxy.h:

Canonical link: https://commits.webkit.org/259829@main
